### DISCUSSION TITLE
Test failed: NextRouter not mounted

### DISCUSSION THREAD

**Discussion by marcos-de-barros-vioque** [neuefische/questions#419] [Web] [answered]
Hello, 

I am trying to highlight in the navbar the active page with `useRouter` (see `Layout.js` component). In addition, I've passed the key to the corresponding component in the `_app.js` in order to remount the component. 
The feature is working correctly in the production environment. However, the corresponding test keeps failing with the following error message: 

`NextRouter was not mounted`

See error details here: https://github.com/marcos-de-barros-vioque/capstone/actions/runs/3741564886/jobs/6351344840

This is the current PR: https://github.com/marcos-de-barros-vioque/capstone/pull/30

Thanks a lot in advance for checking this!

**Comment by actyralf** [comment, marked as answer]
Hey Marcos,

this happens because tests test the components in isolation - context information is not available.
We then have to "mock" the missing context data to pretend there is a context.

In your case the following steps should help:

- Install the `next-router-mock` package: `npm i --save-dev next-router-mock`
- Add these two lines to your `/__tests__/index.test.js` immediately after your imports:
  ```js
  jest.mock("next/router", () => require("next-router-mock"));
  jest.mock("next/dist/client/router", () => require("next-router-mock"));
  ```

**Reply by marcos-de-barros-vioque** [reply]
It's working now. Thanks a bunch, @actyralf !
